### ISSUE TITLE
yarn plug and play and optional peerDependencies

### ISSUE THREAD

**Issue by kapouer** [knex/knex#3059]
Hi,

`yarn --pnp` is a really interesting thing to speedup module loading:
https://yarnpkg.com/en/docs/pnp
however it needs all modules to declare what they depend on to be able to work properly.

In the case of modules like knex, where one of the dependencies is declared by a parent module, the plug and play system fails (e.g. knex won't find pg).
pg, mysql2, sqlite3 (...) should be in peerDependencies, but optional.

This is precisely what yarn's peerDependenciesMeta field do:
https://github.com/yarnpkg/yarn/commit/011a634d787611e9c1f2f1b888212525f1fc096d

However i suppose it needs to be available in npm too before it can be added to knex's package.json.

Meanwhile, a simple solution would be to be able to pass knex the driver module itself, like
```
var pg = require('knex')({
  client: 'pg',
  driver: require('pg'),
  connection: process.env.PG_CONNECTION_STRING,
  searchPath: ['knex', 'public'],
});
```

**Comment by elhigu** [comment]
That is interesting... If it can be setup to package.json without breaking npm functionality, I don't see why it couldn't be added. I don't think that just having `peerDependenciesMeta` attribute there will break anything.

However I still didn't understand how that should work in a way that only needed DB driver package would be included... could you elaborate it more how knex's package.json and parent projects package.json should look in that case?

**Comment by kibertoad** [comment]
Released in 0.20.1
